server/gamenode/GameServer.ts

```
@@ -232,12 +238,30 @@ export class GameServer {
                 ioSocket.disconnect();
                 return;
             }
+            // we get the user from the lobby since this way we can be sure its the correct username.
+            user = lobby.getLobbyUserById(user.id);
             const socket = new Socket(ioSocket);
             lobby.addLobbyUser(user, socket);
+            socket.send('connectedUser', user.id);
             socket.on('disconnect', (_, reason) => this.onSocketDisconnected(user.id, reason));
             return;
         }
-
+        // 2. If user connected to the lobby via a link.
+        if (requestedLobbyId && this.lobbies.has(requestedLobbyId)) {
+            const lobby = this.lobbies.get(requestedLobbyId);
+            const socket = new Socket(ioSocket);
+            if (!user.username) {
+                const newUser = { username: 'Player2', id: user.id };
+                lobby.addLobbyUser(newUser, socket);
+                this.userLobbyMap.set(newUser.id, lobby.id);
+                socket.send('connectedUser', newUser.id);
+                socket.on('disconnect', (_, reason) => this.onSocketDisconnected(user.id, reason));
+                return;
+            }
+            lobby.addLobbyUser(user, socket);
+            this.userLobbyMap.set(user.id, lobby.id);
+            return;
+        }
         // if they are not in the lobby they could be in a queue
```

Not super related but just caught this - if `queuedPlayer` is false, does that mean we're in an error state? If so, we should throw an exception at the end of the method with some debug info (and pass an error back to the user, once we have a framework in place for that)